source/arm11/drivers/gfx.c

```diff
@@ -550,8 +550,47 @@ void GX_processCommandList(const u32 size, const u32 *const cmdList)
 	gx->p3d[GPUREG_CMDBUF_JUMP0] = 1;
 }
 
+void GFX_enterLowPowerState(void)
+{
+	GFX_setForceBlack(true, true);
+	const GfxState *const state = &g_gfxState;
+	LCD_deinit(state->mcuLcdState);
+	stopDisplayControllersSafe();
+
+	for(u8 i = 0; i < 6; i++)
+	{
+		unbindInterruptEvent(IRQ_PSC0 + i);
+	}
+
+	PDN_controlGpu(false, false, false);
+	PDN_sleep();
+}
+
+void GFX_returnFromLowPowerState(void)
+{
+	PDN_wakeup();
+	PDN_controlGpu(true, false, false);
+
+	for(u8 i = 0; i < 6; i++)
+	{
+		bindInterruptToEvent(g_gfxState.events[i], IRQ_PSC0 + i, 14);
+	}
+
+	displayControllerInit(GFX_TOP_2D);
+	const GfxState *const state = &g_gfxState;
+
+	LCD_init(state->mcuLcdState<<1, state->lcdLum);
+	GFX_setForceBlack(false, false);
+
+	GFX_waitForVBlank0();
+	GFX_waitForVBlank0();
+}
+
```

Comment:
I'm a little confused why you added new functions here. GFX_sleep() and GFX_sleepAwake() were supposed to be the functions to call to bring the GFX driver into and out of sleep state.

The problem that PDN stuff is hardcoded in this driver also remains.

Reply:
Thank you so much for your comment. My apologies, I think when playing around, I was trying to first get the old function from the old commit to work, and rebase. I think I forgot to remove these 2 functions. I intend to use the GFX_sleep and GFX_sleepAwake.

d7a30661c975cb398c4f7da021f431d9fb729fc7

Let me try to address your other comments...